src/ec/packages/ec-component-radio/radio-group.html.twig

```diff
@@ -59,44 +56,48 @@
 
 {# Print the result #}
 
-<div
+<fieldset
   class="{{ _css_class }}"
-  role="radiogroup"
-  aria-labelledby="{{ label_id }}"
-  {{ _helper_id is not empty ? "aria-describedby=\"" ~ _helper_id ~ "\""}}
+  {{ _helper_id is not empty ? 'aria-describedby="' ~ _helper_id ~ '"'}}
   {{ _extra_attributes|raw }}
 >
 
   {%- if _label is not empty %}
-    <div
-      class="ecl-form-label{{ hide_label ? ' ecl-form-label--hidden'}}"
-      id="{{ _label_id }}"
+    <legend
+      class="ecl-form-label
+      {{- _invalid ? ' ecl-form-label--invalid' }}"
     >
       {%- block label _label -%}
-    </div>
+      {%- if _required -%}
```

Comment:
The logic here is not valid, we should print the span for the required field only when the required_text is not empty and the optional one in case is not mandatory and the optional_text is not empty.

Reply:
done